include/tvm/support/logging.h

```diff
@@ -113,35 +119,171 @@
 
 namespace tvm {
 
+std::string backtrace();
+
+struct Error : public std::runtime_error {
+  explicit Error(const std::string &s) : std::runtime_error(s) {}
+};
+
+class InternalError : public std::exception {
+ public:
+  InternalError(const std::string& file, int lineno, const std::string& message,
+        const std::time_t& time = std::time(nullptr), std::string stacktrace = backtrace())
```

Comment:
what do we think about pushing this InternalError object into dmlc and defining a weak function backtrace() there that allows you to override/supply a backtrace?

Reply:
I think it makes more sense to keep this error specific to TVM. It prints a message about how to report internal errors to the TVM discuss.